Sources/Ticketmaster/TicketmasterManager.swift

```diff
@@ -13,22 +13,77 @@ class TicketmasterManager {
     let userInfoManager: UserInfoManager
     
     struct Member: Codable {
-        var hostID: String
-        var teamID: String
+        var id: String
+        var email: String?
+        var firstName: String?
+        
+        /// Return the basic UserInfo dictionary for ticketmaster given just the locally known data.
+        var userInfo: [String: String] {
+            return [
+                "email": email,
+                "firstName": firstName
+            ].compactMapValues { $0 }
+        }
     }
     
     var member = PersistedValue<Member>(storageKey: "io.rover.RoverTicketmaster")
     
+    struct LegacyMember: Codable {
+        var hostID: String
+        var teamID: String
+    }
+    var legacyMember = PersistedValue<LegacyMember>(storageKey: "io.rover.RoverTicketmaster")
+    
     init(userInfoManager: UserInfoManager) {
         self.userInfoManager = userInfoManager
+        
+        // do migration from 3.2 and older, if needed.
+        if let legacyData = legacyMember.value, member.value == nil {
+            if !legacyData.hostID.isEmpty {
+                member.value = Member(id: legacyData.hostID, email: nil, firstName: nil)
+                os_log("Migrated Ticketmaster data for Archtics member: %s", log: .general, legacyData.hostID)
+            } else if !legacyData.teamID.isEmpty {
+                member.value = Member(id: legacyData.teamID, email: nil, firstName: nil)
+                os_log("Migrated Ticketmaster data for TM member: %s", log: .general, legacyData.teamID)
+            } else {
+                os_log("Unable to migrate TM data for either type, since neither was present. Ignoring.", log: .general)
+                member.value = nil
+            }
+        }
     }
 }
 
 // MARK: TicketmasterAuthorizer
 
 extension TicketmasterManager: TicketmasterAuthorizer {
     func setCredentials(accountManagerMemberID: String, hostMemberID: String) {
-        self.member.value = Member(hostID: hostMemberID, teamID: accountManagerMemberID)
+        if !accountManagerMemberID.isEmpty {
+            setCredentials(id: accountManagerMemberID, email: nil, firstName: nil)
+        } else if !hostMemberID.isEmpty {
+            setCredentials(id: hostMemberID, email: nil, firstName: nil)
+        } else {
+            // neither value was given, treat it as a logout.
+            os_log("Neither accountManagerMemberID or hostMemberID was given to Rover ticketmaster module legacy API.", log: .general, type: .fault)
+            clearCredentials()
+        }
+    }
+    
+    func setCredentials(id: String, email: String?, firstName: String?) {
+        let newMember = Member(id: id, email: email, firstName: firstName)
+        self.member.value = newMember
```

Comment:
You not checking for empty `id` here?

Reply:
here’s no optionality concept there so far as I know (empty string or otherwise).